dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -67,6 +69,11 @@
 @Entity
 @ApiModel(value = "Version", description = "Base class for versions of entries in the Dockstore")
 @Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+
+@NamedQueries({
+        @NamedQuery(name = "io.dockstore.webservice.core.Version.findVersionInEntry", query = "SELECT v FROM Version v WHERE :entryId = v.parent.id AND :versionId = v.id")
```

Comment:
Should probably put a comment mentioning that this namedQuery is to double check that the user didn't put a workflowId they have access to with a workflowVersionId (from a different hidden workflow) they don't have access to.  

My first thought when looking at this was that the versionId isn't unique and needs the parentId.